### ISSUE TITLE
Incorrect datetime format during bulk insert

### ISSUE THREAD

**Issue by Shaggy84675** [laravel/framework#42561]
- Laravel Version: 9.14.1
- PHP Version: 8.1.2
- Database Driver & Version: 8.0.26 - MySQL Community Server - GPL

### Description:
When doing bulk insert through `Model::insert($array)`, the timestamps are formatted incorrectly and I'm getting this error:

`'SQLSTATE[22007]: Invalid datetime format: 1292 Incorrect datetime value: '2022-05-30T05:39:13.573151Z' for column 'created_at' at row 1 (SQL: insert into `payment_items` (`name`, `created_at`, `updated_at`) values (First item, 2022-05-30T05:39:13.573151Z, 2022-05-30T05:39:24.084337Z))'`

I feel like it also ignores `protected $dateFormat = 'Y-m-d H:i:s.u';` property, what's set in my model.

I think the problem occured in later updates of Laravel 9, because I don't think I had issues with it when Laravel 9 came out. Unless there's other problem what causes it and I missed it.

### Steps To Reproduce:
Because I've tried this in fresh installation of Laravel to make sure it's not issue in my application, you can reproduce it pretty easily in any model what contains any timestamp column:
```php
$items = collect();

$pi = new PaymentItem;
$pi->name = "First item";
$pi->created_at = \Carbon\Carbon::now();
$pi->updated_at = \Carbon\Carbon::now();

PaymentItem::insert($items->toArray());
```

**Comment by driesvints** [comment]
>  8.0.26 - MySQL Community Server - GPL

We don't support this database driver, sorry. Please try one which we do: https://laravel.com/docs/9.x/database#introduction

**Comment by Shaggy84675** [comment]
MySQL is supported, so you mean like version 8? Because in the docs is 5.7+.

**Comment by driesvints** [comment]
Ah sorry, I misread this. `Y-m-d H:i:s.u` isn't a valid timestamp value in MySQL. `Y-m-d H:i:s` is.

**Comment by Shaggy84675** [comment]
Interesting, because I'm using that timestamp value without any issues. Anyway still getting the same error:
![obrazek](https://user-images.githubusercontent.com/1374803/170943182-71ec1b8b-9db2-40ed-abd2-920b510c6369.png)

**Comment by Shaggy84675** [comment]
That's interesting, because I'm using that timestamp value without any issues. Anyway even after the change, I'm still getting the same error:
![obrazek](https://user-images.githubusercontent.com/1374803/170943431-e4d64f90-9ca7-455b-ba50-ba8ed6b6f017.png)

**Comment by Shaggy84675** [comment]
I've found the issue. So after converting the collection into array with $items->toArray() it also converted the timestamp into the ISO-8601 format which is not supported by MySQL. I added this into my model: https://laravel.com/docs/7.x/upgrade#date-serialization and now it works without any issues.

I'm not sure why it worked half year ago when I wrote the script. I thought the issue was somewhere during executing the insert method. That's my bad I didn't check it, but I spent so much time on solving that issue yesterday, that I absolutely forgot it.
